### PR TITLE
Start a contributor guide

### PR DESCRIPTION
**Pull request by rhansen**
Addreses #122, #129

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+To achieve a history like the above, the pull request must be rebased
+onto `master` before merging. Unfortunately, GitHub does not have a
+one-click way to do this (the "Rebase and merge" option does a
+fast-forward merge, which is not what we want). See
+[isaacs/github#1143](https://github.com/isaacs/github/issues/1143) and
+[isaacs/github#1017](https://github.com/isaacs/github/issues/1017). Until
+GitHub adds that feature, it has to be done manually:
+
+```
```
Comment (SuperSandro2000) [inline CONTRIBUTING.md:114]: Adding ```shell should improve coloring.

Comment (rhansen) [reply CONTRIBUTING.md]: Good suggestion, thanks!

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+git checkout -b "${PR_USER:?}-${PR_BRANCH:?}" "${PR_USER:?}/${PR_BRANCH:?}"
+
+# Rebase the commits (optionally using -i to clean up history) onto
+# the current ddclient master branch
+git rebase origin/master
+
+# Force update the contributor's fork. This will only work if the
+# contributor has checked the "Allow edits by maintainers" box in the
+# PR. If not, you will have to manually merge the rebased commits.
+git push -f
```
Comment (wimpunk) [inline CONTRIBUTING.md:143]: Why do you force a push?

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+*   6ae69a1 rhansen@rhansen.org 2020-05-30 22:23:57 -0400
+|\          Merge pull request #141 from ddclient/show-debug-ssl
+| * 096288e sandro.jaeckel@gmail.com 2020-05-30 04:42:27 +0200
+| |         Expand tabs to spaces in vim
+| * 0206262 sandro.jaeckel@gmail.com 2020-05-30 04:40:58 +0200
+|/          Show debug connection settings after evaluating use-ssl
+...
+```
+
+See https://stackoverflow.com/a/15721436 for an explanation of the
```
Comment (wimpunk) [inline CONTRIBUTING.md:103]: I don't believe this is an useful approch if there's only one commit.  Correct me if I'm wrong.

File: CONTRIBUTING.md (hunk 2 above)
Comment (rhansen) [reply CONTRIBUTING.md:143]: Rebasing causes divergence, so Git will reject the push unless forced.

File: CONTRIBUTING.md (hunk 3 above)
Comment (rhansen) [reply CONTRIBUTING.md:103]: It's less useful for sure, but still has value. In particular, the merge commit gets a reference to the pull request. (I think that the "squash and merge" option modifies the commit description to add a reference to the pull request, which would make it even less useful.)

Comment (rhansen) [reply CONTRIBUTING.md:117]: Done.

**Comment by rhansen** [comment]
I haven't heard any feedback on this in a while, so I'm going to merge this later tonight unless anyone objects.

**Pull request merged**